nebula_ros/src/continental/continental_ars548_hw_interface_ros_wrapper.cpp

```diff
@@ -0,0 +1,329 @@
+// Copyright 2023 Tier IV, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "nebula_ros/continental/continental_ars548_hw_interface_ros_wrapper.hpp"
+
+#include <chrono>
+#include <thread>
+
+namespace nebula
+{
+namespace ros
+{
+ContinentalARS548HwInterfaceRosWrapper::ContinentalARS548HwInterfaceRosWrapper(
+  const rclcpp::NodeOptions & options)
+: rclcpp::Node("continental_ars548_hw_interface_ros_wrapper", options),
+  hw_interface_(),
+  diagnostics_updater_(this)
+{
+  if (mtx_config_.try_lock()) {
+    interface_status_ = GetParameters(sensor_configuration_);
+    mtx_config_.unlock();
+  }
+  if (Status::OK != interface_status_) {
+    RCLCPP_ERROR_STREAM(this->get_logger(), this->get_name() << " Error:" << interface_status_);
+    return;
+  }
+  hw_interface_.SetLogger(std::make_shared<rclcpp::Logger>(this->get_logger()));
+  std::shared_ptr<drivers::SensorConfigurationBase> sensor_cfg_ptr =
+    std::make_shared<drivers::ContinentalARS548SensorConfiguration>(sensor_configuration_);
+  hw_interface_.SetSensorConfiguration(
+    std::static_pointer_cast<drivers::SensorConfigurationBase>(sensor_cfg_ptr));
```

Comment:
I think it's cleaner without the implicit cast to `SensorConfigurationBase`:
```suggestion
  std::shared_ptr<drivers::ContinentalARS548SensorConfiguration> sensor_cfg_ptr =
    std::make_shared<drivers::ContinentalARS548SensorConfiguration>(sensor_configuration_);
  hw_interface_.SetSensorConfiguration(
    std::static_pointer_cast<drivers::SensorConfigurationBase>(sensor_cfg_ptr));
```

Reply:
Thanks!